### PR TITLE
Docs: add `py-modindex` to side bar (fix #343)

### LINKED ISSUE #343: docs: add api reference to left navigation pane
[API reference](https://apscheduler.readthedocs.io/en/latest/py-modindex.html) is most useful day-to-day  page (after you pass tutorial). Second time I spent 5 minutes to find link to this page, since it is not on left navigation pane. Sugget to add it.

### PR DESCRIPTION
**Pull request by pohmelie**
fix #343

### CODE REVIEW

**Comment by agronholm** [comment]
Did something appear on the left hand panel for you? Didn't for me.

**Comment by agronholm** [comment]
Actually it seems that using `sphinx_rtd_theme` does make the TOC appear on the side bar. I should probably use that even for local builds by default.

**Comment by pohmelie** [comment]
If this theme is pip instalable, then we can try it localy.

**Comment by pohmelie** [comment]
Yeah, looks like it works without full toc extension. I will fix that.

File: .readthedocs.yml [hunk 1]
```diff
@@ -11,3 +11,4 @@ python:
     - tornado
     - twisted
     - zookeeper
+    - docs
```
Comment (agronholm) [inline .readthedocs.yml]: Rename this to "doc" (I like to stay consistent with my other projects).

File: docs/index.rst [hunk 2]
```diff
@@ -17,6 +17,7 @@ Table of Contents
   contributing
   extending
   faq
+  py-modindex
 
 
 Indices and tables
```
Comment (agronholm) [inline docs/index.rst]: You can remove the second API reference link now.

File: setup.py [hunk 3]
```diff
@@ -61,7 +61,11 @@
         ],
         'testing:python_version == "2.7"': ['mock'],
         'testing:python_version == "3.4"': ['pytest_asyncio < 0.6'],
-        'testing:python_version >= "3.5"': ['pytest_asyncio']
+        'testing:python_version >= "3.5"': ['pytest_asyncio'],
+        'docs': [
```
Comment (agronholm) [inline setup.py]: Rename this to "doc".

**Comment by agronholm** [comment]
Thanks! This has been somewhat of a PITA at times.

**Pull request merged**